tests/testthat/test-randomforest.R

```diff
@@ -190,3 +190,18 @@ test_that("DS-2766: check that output doesn't get too big",
                       show.labels = TRUE, output = "Importance", data = hair100,)
     expect_true(object.size(z) < 9000000)
 })
+
+test_that("DS-2607: RF $ and other non-syntatic names in formula/data set name",
+{
+    set.seed(301)
+    `Cola Tracking - January to September.sav$Variables$d1` <- runif(100)
+    `Cola Tracking - January to September.sav$Variables$d2` <- runif(100)
+    `Cola Tracking - January to September.sav$Variables$d3` <- runif(100)
+    form <- `Cola Tracking - January to September.sav`$Variables$d1 ~
+        `Cola Tracking - January to September.sav`$Variables$d2 +
+        `Cola Tracking - January to September.sav`$Variables$d3
+    expect_error(out <- MachineLearning(algorithm = "Random Forest", formula = form),
+                 NA)
+    expect_equal(out$formula, d1 ~ d2 + d3, check.attributes = FALSE)
+    expect_error(print(out), NA)
+})
```

Comment:
Could perhaps add another test to check the "Questions" string is matched in the calls to `sub` and perhaps also a label that includes escaped backticks? Perhaps this test too?

```
`Cola Tracking - January to September.sav$Questions$\`Long label: \`with ticks\`\`` <- runif(100)
    form <- `Cola Tracking - January to September.sav$Questions$\`Long label: \`with ticks\`\`` ~
      `Cola Tracking - January to September.sav$Variables$d2` +
      `Cola Tracking - January to September.sav$Variables$d3`
    expect_error(out <- MachineLearning(algorithm = "Random Forest", formula = form),
                 NA)
```

Reply:
If you try it out in Q or Displayr, the Outcome and predictor dropboxes only accept variables and the `QFormula(formOutcomeVariable ~ formPredictorVariables)` in the R Code will always expand to `___$Variables$syntaticallyValidName`, since we force variable names to always be syntatically valid. I added the unit test you suggested anyway. A user could still manually edit the R Code.